x-pack/platform/plugins/shared/alerting/server/task_runner/log_alerts.ts

```diff
@@ -90,61 +92,64 @@ export function logAlerts<
     ruleRunMetricsStore.setNumberOfNewAlerts(newAlertIds.length);
     ruleRunMetricsStore.setNumberOfActiveAlerts(activeAlertIds.length);
     ruleRunMetricsStore.setNumberOfRecoveredAlerts(recoveredAlertIds.length);
-    for (const id of recoveredAlertIds) {
-      const alert = recoveredAlerts[id];
-      const { group: actionGroup } = alert.getLastScheduledActions() ?? {};
-      const uuid = alert.getUuid();
-      const state = recoveredAlerts[id].getState();
-      const maintenanceWindowIds = alert.getMaintenanceWindowIds();
-      const message = `${ruleLogPrefix} alert '${id}' has recovered`;
-      alertingEventLogger.logAlert({
-        action: EVENT_LOG_ACTIONS.recoveredInstance,
-        id,
-        uuid,
-        group: actionGroup,
-        message,
-        state,
-        flapping: recoveredAlerts[id].getFlapping(),
-        ...(maintenanceWindowIds.length ? { maintenanceWindowIds } : {}),
-      });
-    }
 
-    for (const id of newAlertIds) {
-      const alert = activeAlerts[id];
-      const { actionGroup } = alert.getScheduledActionOptions() ?? {};
-      const state = alert.getState();
-      const uuid = alert.getUuid();
-      const maintenanceWindowIds = alert.getMaintenanceWindowIds();
-      const message = `${ruleLogPrefix} created new alert: '${id}'`;
-      alertingEventLogger.logAlert({
-        action: EVENT_LOG_ACTIONS.newInstance,
-        id,
-        uuid,
-        group: actionGroup,
-        message,
-        state,
-        flapping: activeAlerts[id].getFlapping(),
-        ...(maintenanceWindowIds.length ? { maintenanceWindowIds } : {}),
-      });
-    }
+    if (autoRecoverAlerts) {
```

Comment:
I was curious if this was going to play in with the other thing we have for auto-recovering alerts, which is I think is only used by security?  But it only deals with the event log?  Shouldn't it be called `shouldLogAlertEventDocs` (or hopefully something shorter!)?

Reply:
Ah, makese sense, I didn't thread it through; don't need another flag (yet).  Ya, let's change the name in this function anyway to make it a little more clear.

Reply:
> I actually wonder if we need to log these event log docs in any circumstance now that we have alerts as data for all rule types. We could just writing the alert UUIDs to the execute doc. I don't know that it gives that much additional information. Maybe a followup?

Ya, would be good actually to find the remaining alert-oriernted APIs still using the event log, like `getAlertSummary()`.  Regardless of whether we'd actually stop generating these docs (which seems fine), it would be good to find all the places dependent on these docs, which should really be changed to use AAD directly.

So ya, let's open an issue.

Reply:
Followup issue: https://github.com/elastic/kibana/issues/237869